server/services/mqtt/lib/handler/devicestate.js

```diff
@@ -0,0 +1,22 @@
+const { EVENTS } = require('../../../../utils/constants');
+
+/**
+ * @description Handle a new message receive in MQTT.
+ * @param {Object} mqttHandler - The Gladys MQTT client.
+ * @param {Object} message - The message sent.
+ * @example
+ * devicestate(gladys.service.mqtt.client, { identifier: 'deviceExternalId', type: 'temp', state: {value: 30} });
+ */
+function devicestate(mqttHandler, message) {
+  // Prepare new state event
+  const event = {
+    device_feature_external_id: `${message.identifier}:${message.type}`,
```

Comment:
I'm not sure this is working, is it?

Here you ask for a "Feature MQTT message key" (not sure if it's understandable..)

<img width="504" alt="Screenshot 2019-09-10 at 08 37 25" src="https://user-images.githubusercontent.com/7365207/64589785-42303c00-d3a6-11e9-8486-920d5f6075f6.png">

And then here, you use "identifier" + "type" 

The type has nothing to do here !

We should just have an identifier like "mqtt:${A_UNIQUE_KEY}" , we let the user choose the second part, and add the `mqtt:` each time (let's not show it on the front side, it's just internal stuff). We could even pre-generate the external_id with the name slugified.

On save, we need to make sure the unique error is handled in case the user use an identifier which already exist, so that a clear message is displayed.

Reply:
The devicestate function is only for backward compatibility with Gladys V3. I have some devices publishing through MQTT with old device state format, I didn't want to change the code of this device. But it's like a proposal. We can force everybody to rewrite MQTT device code to work with V4.

Reply:
I think it's much more complex to provide backward compatibility! The body asked in JSON message are not the same. Let's focus on Gladys 4 I would say, it's not really consistent to handle old messages